server/src/test/java/org/elasticsearch/index/mapper/flattened/FlattenedFieldMapperTests.java

```diff
@@ -176,6 +176,20 @@ public void testNullField() throws Exception {
         assertEquals(0, fields.length);
     }
 
+    public void testBlankFieldName() throws Exception {
+        DocumentMapper mapper = createDocumentMapper(fieldMapping(this::minimalMapping));
+        ParsedDocument parsedDoc = mapper.parse(source(b -> b.field("", "value")));
+        IndexableField[] fields = parsedDoc.rootDoc().getFields("");
```

Comment:
Once the test is updated, I would hope that the root doc actually holds a field with this name, while it previously was rejected during parsing.

Reply:
Done.